services/external/stripe/actions/delete-subscription.ts

```diff
@@ -0,0 +1,44 @@
+import { db } from "@/drizzle/db";
+import { subscriptions, teamMemberships } from "@/drizzle/schema";
+import { and, eq, ne } from "drizzle-orm";
+import { stripe } from "../config";
+
+export async function deleteSubscription(subscriptionId: string) {
+	const subscription = await stripe.subscriptions.retrieve(subscriptionId);
+
+	if (subscription.status === "canceled") {
```

Comment:
I suggest using an early return pattern here to reduce nesting and improve readability:

```
if (subscription.status !== "canceled") {
  return;
}
```

Reply:
updated 42c0a6ddbc0f ♻️